extractors/mysql/extractor_test.go

```diff
@@ -0,0 +1,212 @@
+	})
+
+	t.Run("should return error if no host in config", func(t *testing.T) {
+		extractor := new(mysql.Extractor)
+		_, err := extractor.Extract(map[string]interface{}{
+			"user_id":  user,
+			"password": pass,
+		})
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should return mockdata we generated with mysql running on localhost", func(t *testing.T) {
+		extractor := new(mysql.Extractor)
+		db, err := sql.Open("mysql", "root@tcp(localhost:3306)/")
+		if err != nil {
+			return
+		}
+		err = mockDataGenerator(db)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer cleanDatabase(db)
+		result, err := extractor.Extract(map[string]interface{}{
+			"user_id":  user,
+			"password": pass,
+			"host":     "localhost:3306",
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		expected := getExpectedVal()
+		assert.Equal(t, result, expected)
+	})
+}
+
+func getExpectedVal() (expected []map[string]interface{}) {
+	expected = []map[string]interface{}{
+		{
+			"columns": []map[string]interface{}{
+				{
+					"data_type":   "int",
+					"field_desc":  "",
+					"field_name":  "applicant_id",
+					"is_nullable": "YES",
+					"length":      0,
+				},
+				{
+					"data_type":   "varchar",
+					"field_desc":  "",
+					"field_name":  "first_name",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "varchar",
+					"field_desc":  "",
+					"field_name":  "last_name",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+			},
+			"database_name": "mockdata_meteor_metadata_test",
+			"table_name":    "applicant",
+		},
+		{
+			"columns": []map[string]interface{}{
+				{
+					"data_type":   "varchar",
+					"field_desc":  "",
+					"field_name":  "department",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "varchar",
+					"field_desc":  "",
+					"field_name":  "job",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "int",
+					"field_desc":  "",
+					"field_name":  "job_id",
+					"is_nullable": "YES",
+					"length":      0,
+				},
+			},
+			"database_name": "mockdata_meteor_metadata_test",
+			"table_name":    "jobs",
+		},
+	}
+	return
+}
+
+func mockDataGenerator(db *sql.DB) (err error) {
+	_, err = db.Exec(fmt.Sprintf("DROP DATABASE IF EXISTS %s", testDB))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", testDB))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf("USE %s;", testDB))
+	if err != nil {
+		return
+	}
+	table1 := "applicant"
+	var1 := "(applicant_id int, last_name varchar(255), first_name varchar(255))"
+	table2 := "jobs"
+	var2 := "(job_id int, job varchar(255), department varchar(255))"
+	tableQuery := "CREATE TABLE "
+	err = createTable(tableQuery, table1, var1, db)
+	if err != nil {
+		return
+	}
+	err = createTable(tableQuery, table2, var2, db)
+	if err != nil {
+		return
+	}
+	_, err = db.Exec("CREATE USER IF NOT EXISTS 'user'@'%' IDENTIFIED BY 'pass';")
```

Comment:
we could use our `user` and `pass` variables to make sure everything is in sync.